### DISCUSSION TITLE
About Custom Renderer

### DISCUSSION THREAD

**Discussion by noo-have** [solidjs/solid#1349] [Q&A]
I encountered such a problem:
```tsx
const A = () => <s/ >
const B = () => A()
const C = () => B()
console.log(A(), B(),C())
```
If I do this, the result is in line with expectations;
But if I write:
```tsx
const A = () => <s/ >
const B = () => <A/ >
const C = () => <B/ >
console.log(A(),B(),C())
```
The console starts printing 3 'readSignals'!
More maddening, I need to do this to get the right results:
```tsx
console.log(A(), B()(), C()()())
```
What is the difference between 'A ()' and '\<A\/\>'? What should I do?

**Comment by atk** [comment]
The playground output shows how these JSX expressions are converted:

```tsx
const B = () => <A />;
// is converted to
const B = () => createComponent(A, {});
```
Calling A() is not converted.

**Reply by ryansolid** [reply]
And the difference is `createComponent` wraps in an `untrack`. Although I wonder if the weirdness is dev tools doing extra wrapping or something like that. Because we shouldn't be getting extra function calls.
